pkg/collector/resource.go

```diff
@@ -215,6 +223,52 @@ func resourceMatchesNamespace(res unstructured.Unstructured, namespace *string)
 	return true
 }
 
+func resourceMatchesAnnotations(res unstructured.Unstructured, annotations map[string]string) bool {
+	if annotations != nil {
+		resAnnotations := res.GetAnnotations()
+
+		for key, val := range annotations {
+			v, ok := resAnnotations[key]
+			if !ok {
+				return false
+			}
+
+			match, err := regexp.MatchString(`^`+val+`$`, v)
+			if err != nil {
+				klog.Error(err)
+			}
+			if !match {
+				return false
+			}
+		}
+	}
+
+	return true
+}
+
+func resourceMatchesLabels(res unstructured.Unstructured, labels map[string]string) bool {
+	if labels != nil {
+		resLabels := res.GetLabels()
+
+		for key, val := range labels {
+			v, ok := resLabels[key]
+			if !ok {
+				return false
+			}
+
+			match, err := regexp.MatchString(`^`+val+`$`, v)
+			if err != nil {
+				klog.Error(err)
+			}
+			if !match {
+				return false
+			}
+		}
+	}
+
+	return true
+}
+
```

Comment:
I feel like this could be easily reduced to one function accepting a two maps. labels and annotations both are string maps.